Fix typos in messages translations, correct escaping of strings

echoix opened the pull request:
I suggest here another set of changes for the messages/*.js files. Mainly, uniformize the use of the official "Node-RED" spelling vs Node-Red, JavaScript instead of Javascript, formatting the files to remove blank spaces at the end of lines, fix the escaping of the `\\n` to display `\n` in the UI.

I will also add some comments of things I find weird within this PR.

echoix commented on messages/en.js:
```diff
@@ -13,35 +13,35 @@ Blockly.Msg.NODE_OBJECT_SET_JSON = "set %1 property %2 to %3";
 Blockly.Msg.NODE_FLOW_MEMORY = "flow";
-Blockly.Msg.NODE_FLOW_MEMORY_TOOLTIP = "Access the data in the Node-Red flow memory.";
+Blockly.Msg.NODE_FLOW_MEMORY_TOOLTIP = "Access the data in the Node-RED flow memory.";
 Blockly.Msg.NODE_CONTEXT_MEMORY = "(node)context";
```
What is `(node)context` exactly? Should there be a space? Translations carry on this term literally.

bartbutenaers replied:
The (node)context is simply the context memory of a node.  As you can see [here](https://nodered.org/docs/user-guide/context) you have node context, flow context, and global context.  In case of flow and global, the 'context' is not mentioned in our blockly toolbar.  After a discussion on Discourse at the time being, it was decided that to to it like that.  Personally I found it more readable if the 'context' was mentioned in all 3 blocks...

cymplecy replied:
Node-RED isn't the clearest when it comes to naming things - worst culprit is the over used term "flow" :)
Node-RED just calls the node local storage object "context" whereas it calls the others "global context" and "flow context" so I suggested using "(node) context" to try to make things clearer but using the ( ) to show that it wasn't official Node-RED name.

echoix replied:
Ok, so the missing space was a typo, so I used this assumption to edit.

echoix commented on messages/en.js:
```diff
@@ -100,13 +100,13 @@ Blockly.Msg.OBJECT_KEYS_TOOLTIP = "Get all keys from the object";
 Blockly.Msg.TEXT_SPECIAL_CHARACTER_CR = "carriage return (\\r)";
 Blockly.Msg.TEXT_SPECIAL_CHARACTER_CRLF = "carriage return line feed (\\r\\n)";
 Blockly.Msg.TEXT_SPECIAL_CHARACTER_TAB = "tab (\\t)";
 Blockly.Msg.TIMER_CONDITIONAL = "Repeat every %1 seconds, until %2 %3";
```
For the text of a block (the operation of the block), should the text/sentence start with capital? Here is an example with capitalization, while most of the other block strings start with a lower-case. What is used with the blocks coming from the original Blockly if we want to match the style? Once the capitalization style is chosen, we could uniformize the translations, like in French.

bartbutenaers replied:
Seems that the standard blocks from Google indeed have a capital at the start of their tooltips:

![image](https://user-images.githubusercontent.com/14224149/130400904-12950537-f522-4237-be1f-762852eaa759.png)

If you have time, you can change that.  But I have to continue with other stuff on my backlog....

echoix replied:
I uniformized it in 34c153562da0813f8137e43720f3a91f94bdc85c

echoix commented on messages/en.js:
```diff
@@ -138,16 +138,16 @@ Blockly.Msg.DATETIME_CONVERT_FROM_DATE = "get %1 from timestamp %2";
 Blockly.Msg.DATETIME_CONVERT_FROM_DATE_TOOLTIP = "Get the specified information from a timestamp";
```
Should all the tooltips end with a period? If so, we could uniformize at least the French translation

bartbutenaers replied:
Seems that indeed the standard Blockly blocks (offered by Google) have a period at the end of their tooltips:

![image](https://user-images.githubusercontent.com/14224149/130400590-5a074145-8cbd-4f89-a642-e9fbf02a5539.png)

So if you like you can change that.

echoix replied:
I uniformized it in 34c153562da0813f8137e43720f3a91f94bdc85c

echoix commented on messages/en.js:
```diff
@@ -100,13 +100,13 @@ Blockly.Msg.OBJECT_KEYS_TOOLTIP = "Get all keys from the object";
 Blockly.Msg.TEXT_SPECIAL_CHARACTER = "special character %1";
-Blockly.Msg.TEXT_SPECIAL_CHARACTER_TOOLTIP = "A special character that shouldn't be escaped (by a '\' prefix";
+Blockly.Msg.TEXT_SPECIAL_CHARACTER_TOOLTIP = "A special character that shouldn’t be escaped (by a '\\' prefix)";
 Blockly.Msg.TEXT_SPECIAL_CHARACTER_LF = "line feed (\\n)";
```
In the Japanese translation, there was a good idea to add in parentheses (LF), (CR), (CRLF), before the (\n), (\r), (\r\n), since it is the ASCII symbol name. However, it adds more information, whereas Blockly is intended to be really simple. It's an open discussion here

bartbutenaers replied:
It makes sense to me to add that, since it are ASCII symbol names like you said.
Unless @cymplecy has an objection for some reason?

cymplecy replied:
Fine with me

echoix replied:
Did it.

bartbutenaers commented:
@echoix,

Could you please explain the following fix a bit in more detail?

> fix the escaping of the \\n to display \n in the UI

Because at first sight it seems to me that everything is already correctly displayed:

![image](https://user-images.githubusercontent.com/14224149/130401398-80ade49f-222a-44d4-b61b-0ac2059d7ad7.png)

Or not?

echoix commented:
**English**
Original English:
![image](https://user-images.githubusercontent.com/27212526/130449698-1e791a55-8d6a-4627-a4a2-ae6688b40614.png)
![image](https://user-images.githubusercontent.com/27212526/130449662-1a7dde83-4a98-4fc9-b5c3-b5e7ffd5c277.png)
Fixed English:
![image](https://user-images.githubusercontent.com/27212526/130451636-03cba7b8-0050-491e-9e76-c706e5abbb59.png)
![image](https://user-images.githubusercontent.com/27212526/130451921-db6e28a0-7e54-41d8-abce-eea1badae576.png)


**French**
Original French: (not translated, stays in English)
![image](https://user-images.githubusercontent.com/27212526/130449176-07fdb090-b56a-43ff-9200-73e99af89581.png)
![image](https://user-images.githubusercontent.com/27212526/130449363-7301f1f5-fbd9-416f-939f-d15f77685594.png)
Fixed French:
![image](https://user-images.githubusercontent.com/27212526/130452348-a1b3562b-aba0-438f-aa08-89fa96e62ad1.png)
![image](https://user-images.githubusercontent.com/27212526/130452566-bf9a7c16-e812-43b0-b442-1923c1f87771.png)



**Japanese**
Original Japanese: 
![image](https://user-images.githubusercontent.com/27212526/130449002-71c7c860-818f-4626-9e69-cab382908f1b.png)
![image](https://user-images.githubusercontent.com/27212526/130448920-75967fe0-3674-457a-b0bd-b11cfb29ef52.png)
Fixed Japanese:
![image](https://user-images.githubusercontent.com/27212526/130452995-ce74b61e-7bb8-44f3-a98d-bcaad6ea0b75.png)
![image](https://user-images.githubusercontent.com/27212526/130453238-3a78fda0-32b2-4480-b7cb-93103ed681bf.png)



**Dutch**
Original Dutch: 
![image](https://user-images.githubusercontent.com/27212526/130448773-e0c3fef1-6439-474a-8658-ed175b25cb24.png)
![image](https://user-images.githubusercontent.com/27212526/130448650-c70d8e3f-2d39-4473-940a-d7b0dca77a96.png)
Fixed Dutch:
![image](https://user-images.githubusercontent.com/27212526/130453871-6cbf13f9-9739-4ae3-b9e8-e753c2063dbd.png)
![image](https://user-images.githubusercontent.com/27212526/130453952-805d5070-20d0-41eb-a845-d13f7e8ab695.png)



**Russian**
Original Russian:
![image](https://user-images.githubusercontent.com/27212526/130448464-94075577-bce5-46b7-8217-7fb9adf1949f.png)
![image](https://user-images.githubusercontent.com/27212526/130448282-652e1998-aa5f-47d3-bcf9-6b2f4a53a440.png)
Fixed Russian:
![image](https://user-images.githubusercontent.com/27212526/130454152-df53cadf-444c-4c3f-a789-ebe0ffdb453c.png)
![image](https://user-images.githubusercontent.com/27212526/130454214-c656e3fa-838f-44c7-8ead-2ef27ba6e72c.png)

echoix commented on messages/en.js:
```diff
@@ -118,14 +118,14 @@ Blockly.Msg.SWITCH_CASE_TOOLTIP = "When a specific value is matched.";
-Blockly.Msg.MISC_JAVASCRIPT= "Javascript statement %1;";
-Blockly.Msg.MISC_JAVASCRIPT_TOOLTIP = "Enter Javascript code when no blocks offer certain functionality.";
-Blockly.Msg.MISC_JSEXPRESS= "Javascript expression %1";
+Blockly.Msg.MISC_JAVASCRIPT = "JavaScript statement %1;";
```
Is the semicolon inside the string wanted, or is a typo?

bartbutenaers replied:
We use always a semicolon at the end of our JavaScript expressions and statement, even in our translated texts it seems 😉
No you are right: those can be removed here.

cymplecy replied:
Hang on  - isn't it there to show that the user doesn't have to put one in?

echoix replied:
I left it as is for now.

bartbutenaers replied:
Ok fine for me!

echoix commented:
And maybe one last thing that I wasn't able to verify. I couldn't find in the UI where the strings TIMER_CONDITIONAL, TIMER_COUNTING and their tooltips were used. It seems that they are used in `lib/extra/extraBlocksDefs.js`, but the comments on these two files might seem to indicate that they are hidden for 3 years.
https://github.com/bartbutenaers/node-red-contrib-blockly/blob/5c9aef425c25718afc1632b893dd1f980c096703/lib/extra/extraBlocksCodeGen.js#L1-L27
https://github.com/bartbutenaers/node-red-contrib-blockly/blob/5c9aef425c25718afc1632b893dd1f980c096703/lib/extra/toolbox.xml#L2-L7

bartbutenaers commented:
Seems that are two blocks to create an interval timer.
+ One block executes the timer a specified number of times.
+ One block executes the timer until a condition becomes true.

I might imagine that it would be nice to add both blocks in the future to the "Timers" category, as an extra alternative to the setInterval block (because they will result in less blocks for those specific use cases).  After I have redesigned both blocks to work together with the clearInterval block.

So I would keep the texts.  Any thoughts @cymplecy ?

cymplecy commented:
No thoughts - brain is hurting too much :)

echoix commented:
I think we'll let the strings there for sure, since the blocks that use them are still there and implemented. It was only a matter of knowing why they weren't available to use.

bartbutenaers commented:
@echoix ,
Thanks a lot for this full review of all our translations!!!!!
Bart
